pkg/internal/codegen/parse/util.go

```diff
@@ -326,3 +357,32 @@ func checkType(props *v1beta1.JSONSchemaProps, s string, enums *[]v1beta1.JSON)
 		*enums = append(*enums, v1beta1.JSON{Raw: []byte(`"` + s + `"`)})
 	}
 }
+
+// Scale subresource requires specpath, statuspath, selectorpath key values, represents for JSONPath of
+// SpecReplicasPath, StatusReplicasPath, LabelSelectorPath separately. e.g.
+// +kubebuilder:subresource:scale:specpath=.spec.replica,statuspath=.status.replica,selectorpath=
+func parseJSONPath(t *types.Type) (map[string]string, error) {
+	jsonPath := make(map[string]string)
+	for _, c := range t.CommentLines {
+		if strings.Contains(c, "+kubebuilder:subresource:scale") {
+			paths := strings.Split(c, ":")[3:][0]
+			path := strings.Split(paths, ",")
+			if len(path) != 3 {
+				return nil, fmt.Errorf(jsonPathError)
+			}
+			for _, s := range path {
+				kv := strings.Split(s, "=")
+				if kv[0] == specReplicasPath || kv[0] == statusReplicasPath || kv[0] == labelSelectorPath {
+					jsonPath[kv[0]] = kv[1]
+				}
```

Comment:
else part is error. If we do that, then we don't need the second loop below. We should add unit test to verify the logic.